src/index.js

```diff
@@ -105,31 +109,32 @@ export class SlideDeck extends React.Component {
     }
 
     if (e.metaKey || e.ctrlKey || e.shiftKey) return
+
     const alt = e.altKey
 
     switch (e.keyCode) {
       case keys.right:
       case keys.space:
         e.preventDefault()
-        this.update(inc)
+        this.next(this.state)
```

Comment:
Any reason to switch away from the functional set state approach here?

Reply:
Wanted to decouple the event handling from the actual state updating. Moved `this.update(...)` into `next`.
But yes, no need to pass the current state, we can access it inside `next` directly.